code/datums/flock/flockprotection.dm

```diff
@@ -1,76 +1,83 @@
+
+/// Mark a specific flock as interested in this
+/datum/component/flock_interest
+	/// The flock who is intently interested in this thing.
+	var/datum/flock/flock
+
+/datum/component/flock_interest/Initialize(datum/flock/flock)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.flock = flock
+
+/datum/component/flock_interest/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_FLOCK_ATTACK, .proc/handle_flock_attack)
+
+/// If flockdrone is in our flock, deny the attack, otherwise scream and cry
+/datum/component/flock_interest/proc/handle_flock_attack(source, mob/user, var/intentional)
+	if(!istype(user))
+		return
+
+	var/mob/living/critter/flock/F = user
+	if (istype(F) && F.flock == src.flock)
+		if(intentional)
+			boutput(user, "<span class='alert'>The grip tool refuses to harm this, jamming briefly.</span>")
+		return intentional
+
+	var/mob/living/critter/flock/drone/snitch
+	for (var/mob/living/critter/flock/drone/FD in view(7, source))
```

Comment:
This seems like it could it could get expensive if a lot of flock stuff is being damaged at once next to a pile of drones, not sure how much of a problem it is, but I don't think it would be too bad if flock buildings just self reported without a drone.

Reply:
These are not for buildings (which already report), this is for chairs, windows, doors, etc. It makes little sense for the flock to get angry at you for hitting a chair if they never saw you do it